File: app/src/main/java/com/mtime/player/MemoryPlayRecorder.java
package com.mtime.player;

import android.util.SparseIntArray;

import com.kk.taurus.playerbase.log.PLog;

/**
 * Created by mtime on 2018/1/6.
 */

public class MemoryPlayRecorder {

    static final String TAG = "MemoryPlayRecorder";

    private static final SparseIntArray mRecorders = new SparseIntArray();

    private static int mRecorderPlayIndex;

    public static void recordPlayTime(int id, int time){
        mRecorders.put(id, time);
        PLog.d(TAG,"recordPlayTime : id = " + id + ", time = " + time);
    }

    public static int getRecordPlayTime(int id){
        PLog.d(TAG,"getRecordPlayTime : id = " + id + ", record = " + mRecorders.get(id));
        return mRecorders.get(id);
    }

    public static void recordPlayIndex(int index){
        mRecorderPlayIndex = index;
    }

    public static int getRecorderPlayIndex(){
        return mRecorderPlayIndex;
    }

}

File: player/src/main/java/com/kotlin/android/player/MemoryPlayRecorder.kt
package com.kotlin.android.player

import android.util.SparseIntArray
import com.kk.taurus.playerbase.log.PLog

/**
 * create by lushan on 2020/8/31
 * description:
 */
class MemoryPlayRecorder {
    companion object{
        const val TAG = "MemoryPlayRecorder"

        private val mRecorders = SparseIntArray()

        private  var mRecorderPlayIndex = 0
        fun recordPlayTime(id: Int, time: Int) {
           mRecorders.put(id, time)
            PLog.d(TAG, "recordPlayTime : id = $id, time = $time")
        }

        fun getRecordPlayTime(id: Int): Int {
            PLog.d(TAG, "getRecordPlayTime : id = " + id + ", record = " + mRecorders.get(id))
            return mRecorders.get(id)
        }

        fun recordPlayIndex(index: Int) {
            mRecorderPlayIndex = index
        }

        fun getRecorderPlayIndex(): Int {
            return mRecorderPlayIndex
        }

    }
}